File: data-jpa/src/main/java/top/bettercode/summer/data/jpa/querydsl/QuerydslRepository.java
package top.bettercode.summer.data.jpa.querydsl;

import org.springframework.data.querydsl.QuerydslPredicateExecutor;
import org.springframework.data.repository.NoRepositoryBean;
import top.bettercode.summer.data.jpa.BaseRepository;

/**
 * @param <T>  T
 * @param <ID> ID
 * @author Peter Wu
 */
@NoRepositoryBean
public interface QuerydslRepository<T, ID> extends BaseRepository<T, ID>,
    QuerydslPredicateExecutor<T>, RecycleQuerydslPredicateExecutor<T> {

}

File: data-jpa/src/main/kotlin/top/bettercode/summer/data/jpa/querydsl/QuerydslRepository.kt
package top.bettercode.summer.data.jpa.querydsl

import org.springframework.data.querydsl.QuerydslPredicateExecutor
import org.springframework.data.repository.NoRepositoryBean
import top.bettercode.summer.data.jpa.BaseRepository

/**
 * @param <T>  T
 * @param <ID> ID
 * @author Peter Wu
</ID></T> */
@NoRepositoryBean
interface QuerydslRepository<T, ID> : BaseRepository<T, ID>, QuerydslPredicateExecutor<T>, RecycleQuerydslPredicateExecutor<T> 